pgcli/pgcompleter.py

```diff
@@ -341,6 +369,55 @@ def get_column_matches(self, suggestion, word_before_cursor):
 
         return self.find_matches(word_before_cursor, flat_cols, meta='column')
 
+    def get_join_condition_matches(self, suggestion, word_before_cursor):
+        lefttable = suggestion.parent or suggestion.tables[-1]
+        scoped_cols = self.populate_scoped_cols(suggestion.tables)
+
+        def make_cond(tbl1, tbl2, col1, col2):
+            prefix = '' if suggestion.parent else tbl1 + '.'
+            return prefix + col1 + ' = ' + tbl2 + '.' + col2
+
+        # Tables that are closer to the cursor get higher prio
+        refprio = dict((tbl.ref, num) for num, tbl
+            in enumerate(suggestion.tables))
+        # Map (schema, tablename) to tables and ref to columns
+        tbldict = defaultdict(list)
+        for t, cs in scoped_cols.items():
+            tbldict[(t.schema, t.name)].append(t)
+        refcols = dict((t.ref, cs) for t, cs in scoped_cols.items())
+        # For each fk from the left table, generate a join condition if
+        # the other table is also in the scope
+        conds = [(make_cond(lefttable.ref, rtbl.ref, lcol.name, rcol[2]),
```

Comment:
This list comprehension is a beast! Should probably switch to a more traditional nested for-loop structure